pyomo/solvers/tests/checks/test_multi.py

```diff
@@ -0,0 +1,96 @@
+import pyutilib.th as unittest
+from pyomo.environ import *
+
+
+__author__ = "Sunjeev Kale <https://github.com/sjkale>"
+
+
+#due to stochastic nature of the random restarts, these tests just demonstrate,
+#that for a small sample, the test will not do worse than the standard solver.
+#this is non-exhaustive due to the randomness
+class MultistartTests(unittest.TestCase):
+
+    #test standard random restarts
+    def test_as_good_with_iteration_rand(self):
+        #initialize model with data
+        m = build_model()
+
+        #create ipopt solver
+        optsolver = SolverFactory('ipopt')
+        optsolver.solve(m);
+        for i in xrange(20):
+
+
+            m2 = build_model()
+            optsolver = SolverFactory('multistart')
+            optsolver.solve(m2,iterations = 10)
+
+            self.assertTrue(value(m2.obj) >= value(m.obj))
+            del m2
+
+    #test that other strategies do no worse
+    def test_as_good_with_iteration_other_strategies(self):
+        #initialize model with data
+        m = build_model()
+
+        #create ipopt solver
+        optsolver = SolverFactory('ipopt')
+        optsolver.solve(m);
+        for i in xrange(10):
+
+
+            m2 = build_model()
+            optsolver = SolverFactory('multistart')
+            optsolver.solve(m2,strategy = 'rand_distributed')
+
+            self.assertTrue(value(m2.obj) >= value(m.obj))
+            del m2
+        for i in xrange(10):
+
+
+            m2 = build_model()
+            optsolver = SolverFactory('multistart')
+            optsolver.solve(m2,strategy = 'midpoint_guess_and_bound')
+
+            self.assertTrue(value(m2.obj) >= value(m.obj))
+            del m2
+        for i in xrange(10):
+
+
+            m2 = build_model()
+            optsolver = SolverFactory('multistart')
+            optsolver.solve(m2,strategy = 'rand_guess_and_bound')
+
+            self.assertTrue(value(m2.obj) >= value(m.obj))
+            del m2
+    #test that the high confidence stopping rule with very lenient parameters
+    #does no worse.
+    def test_as_good_with_HCS_rule(self):
+        #initialize model with data
+        m = build_model()
+
+        #create ipopt solver
+        optsolver = SolverFactory('ipopt')
+        optsolver.solve(m);
+        for i in xrange(5):
+
+
+            m2 = build_model()
+            optsolver = SolverFactory('multistart')
+            optsolver.solve(m2,iterations = -1,HCS_param=(.99,.99))
+
+            self.assertTrue(value(m2.obj) >= value(m.obj))
+            del m2
+
+#simple non-convex model with many multiple minima
+def build_model():
```

Comment:
You should run these tests on a few different toy models with different features (Constraints, multiple variables, fixed variables, variables with no bounds, objectives to be minimized, etc.). In particular, make sure to include test cases that exercise any of the logic that you use in your solver class that checks for different model attributes.